### ISSUE TITLE
Add more precise directions on how to code/test in the Contributing documentation

### ISSUE THREAD

**Issue by raphaelfournier** [TailorDev/Watson#238]
I am willing to contribute to Watson. I followed the instructions currently in the http://tailordev.github.io/Watson/contributing/hack page, everything is OK (forking, virtualenv, branch creation, tox).
However, I am missing some directions on how to work at the `Now you can make your changes locally.` step. 

I have a watson installed on my machine, and the development version. What files from my development repo should I edit? I guess they are in the `watson` subfolder. I see a  `build/lib/watson` subfolder, I guess I have to use `make` or something so that files from `watson/` are moved to `build/lib/watson`: which command(s) do(es) that?

Also, once I will have made some changes (and *built* them?), how do I test the new watson behavior/command?  

For instance, I get this error:
```
$ python watson/watson.py
Traceback (most recent call last):
  File "watson/watson.py", line 18, in <module>
    from .config import ConfigParser
ModuleNotFoundError: No module named '__main__.config'; '__main__' is not a package
```
I would also like to use some frames from my production watson, to test some features, but would not like to mess them up. How may I copy some of them so that they are used by the development version?

**Comment by jmaupetit** [comment]
Hi @raphaelfournier!

You are right: the documentation needs a revamp to easily start working on the project. I'll try to answer your questions:

> What files from my development repo should I edit? I guess they are in the `watson` subfolder.

Yes, core files from the project are under the `watson/` directory.

> I see a build/lib/watson subfolder, I guess I have to use make or something so that files from watson/ are moved to build/lib/watson: which command(s) do(es) that?

The `build` directory is created while trying to build (or install) Watson with the setup script, _e.g._:

```bash
$ python setup.py build
# or
$ python setup.py install
```
Files from the `build` directory are temporary and their modification is not intended in the development workflow. You should focus on the `watson` directory.

> Also, once I will have made some changes (and built them?), how do I test the new watson behavior/command?

You should work in a python virtualenv; once activated (and Watson installed in this virtualenv) the `watson` command should point to this development installation:

```bash
# Create a virtualenv
$ python3 -m venv venv
# Activate this virtualenv
$ source venv/bin/activate
# Install Watson's dependencies in this virtualenv
(venv) $ pip install -r requirements-dev.txt
# Install Watson in development mode
(venv) $ python setup.py develop
# Use freshly installed development version of Watson
(venv) $ venv/bin/watson --help
```
Remarks:

1. We have a `Makefile` target for the development installation: `make install-dev` (instead of `pip install -r requirements-dev.txt && python setup.py develop`
2. The `watson` command from your virtualenv should be first in your `PATH` but for some obscure reasons, you should force the relative path to it (in your virtualenv: `venv/bin/watson`) to prevent your shell from calling the system install of Watson.

> I would also like to use some frames from my production watson, to test some features, but would not like to mess them up. How may I copy some of them so that they are used by the development version?

To work with a copy of real data or create fake ones, you will need to define the `WATSON_DIR` environment variable (see [Watson's user guide](http://tailordev.github.io/Watson/user-guide/configuration/#application-folder)). A good idea would be to copy production files to a working directory from your local clone:

```bash
# For GNU/Linux system
(venv) $ cp -R ~/.config/watson data
# Override Watson's directory for this command
(venv) $ WATSON_DIR=$PWD/data venv/bin/watson status
# or override watson's directory for the current shell session
(venv) $ export WATSON_DIR=$PWD/data
```
:point_up: alternatively you can modify the `venv/bin/activate` script to add the `WATSON_DIR` definition:

```bash
WATSON_DIR=$PWD/data
export WATSON_DIR
```

But this definition will be lost if you have to reset (delete) your project's virtualenv.

---

I really hope these information will help you to start hacking with Watson :raised_hands: 

Last remark: as you might not be the one that get stuck after the "Now you can make your changes locally" step, do you mind contributing to Watson with a first documentation-related PR that includes parts of what as been described here?

**Comment by raphaelfournier** [comment]
Hey, thank you very much @jmaupetit!
I will soon* try to implement your suggestions and start hacking on watson. I have not yet deeply looked at the problem, but when running one of your commands I saw that my dev install of watson was in ~/.virtualenvs/watson, because I had a global config to centralize my venvs somewhere (and did not remember that before asking).

I will definitely start by doing some improvements in the documentation with my findings, and submit my first PR :-) 

* soon: probably before the end of this year.

**Comment by raphaelfournier** [comment]
Hello @jmaupetit,
I have done the work for the documentation part, and submitted a PR. I hope that it is okay. If not, do not hesitate to disapprove it, I'll make another one.
Thanks again for your comments, it helped me a lot ! 

BTW, I had some problems with the documentation part, since my system was in Python 3.7 and mkdocs was in 0.14, which are incompatible. I upgraded mkdocs to 1.0.4, works like a charm, after also including the theme packages (`pip install --upgrade mkdocs-bootstrap mkdocs-bootswatch`). I included that info in the new hack.md.
